services/net/contentmigration/migrators/ContentMigratorFactoryExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using TNO.Services.ContentMigration.Config;
+
+namespace TNO.Services.ContentMigration.Migrators;
+
+/// <summary>
+///
+/// </summary>
+public static class ContentMigratorFactoryExtensions
+{
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="services"></param>
+    /// <param name="configuration"></param>
+    /// <returns></returns>
+    public static IServiceCollection AddSourceMigrators(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.Configure<MigratorOptions>("ClipMigrator", configuration.GetSection("Service:ClipMigrator"));
```

Comment:
I don't see anything obvious referencing the names you provide.  Is there a convention that understands how to inject the correct object?

Reply:
The name I'm using as a convention is the same as the name of the class which will need to consume those named options.  If you take a look at `ContentMigrator.cs` line 68 you can see it there.  I have added a comment for it now.
```
        // as a convention the Named Option should be the same as the class name which needs to consume it
        this.MigratorOptions = migratorOptions.Get(this.GetType().Name);
```

Reply:
arguably I could wire it a bit tighter with something like this:
```
services.Configure<MigratorOptions>(typeof(ClipMigrator).Name, configuration.GetSection("Service:ClipMigrator"));
```